Add gsoc AI and LLM usage guidelines (#1167)
OWASP/www-community: pages/initiatives/gsoc/gsoc_ai_usage_guidelines.md

---
title: "AI and LLM Usage Guidelines"
description: "Guidelines for responsible and transparent use of AI and Large Language Models (LLMs) by Google Summer of Code contributors."
---

# LLM Usage Policy

> ## Purpose

Reviewer: ```suggestion
## NOTE

⚠️  Failure to disclose LLM usage or submission of unreviewed AI-generated content **may result in proposal dismissal** from GSoC without further review ⚠️ 

## Purpose
```

> This policy establishes guidelines for the responsible and transparent use of **AI and Large Language Models (LLMs)** (e.g., ChatGPT, Gemini, Claude, Copilot) by **Google Summer of Code (GSoC)** contributors working on open-source projects.

Reviewer: ```suggestion
This document establishes guidelines for the responsible and transparent use of **AI and Large Language  Models (LLMs)** (e.g., ChatGPT, Gemini, Claude, Copilot, etc) by **Google Summer of Code (GSoC)** contributors working on OWASP projects.
```

The intent of this guideline is to support learning and productivity while preserving originality, accountability, and open-source integrity.

---

## 1. Acceptable Use

Contributors **may use LLMs** for supportive tasks, including:

- Improving clarity, grammar, or structure of documentation or reports
- Assisting with code comments, summaries, or test-case generation
- Learning and exploration, provided outputs are independently reviewed and understood

LLMs **must not** be used to automatically generate substantial portions of code, documentation, designs, or research outputs **without full review and acknowledgment** by the contributor.

All final submissions must reflect the contributor’s own understanding, decision-making, and work.

---

## 2. Disclosure Requirement

> If an LLM was used in any part of a proposal, report, or project artifact, contributors **must include** a section titled **“LLM Usage Considerations”**.

Reviewer 2: ```suggestion
If an LLM was used in any part of a proposal, report, or project artifact, contributors **must include** a section titled **"LLM Usage Considerations"**.
```

This section should:

- Describe **how and why** an LLM was used  
  (e.g., proofreading documentation, suggesting variable names)
- Confirm that all generated content was **independently reviewed, tested, and verified**
- Include the following statement:

> > “LLMs were used for supportive or editorial purposes in this work. All outputs were reviewed and validated by the contributor to ensure accuracy and originality.”

Reviewer 2: ```suggestion
> LLMs were used for supportive or editorial purposes in this work. All outputs were reviewed and validated by the contributor to ensure accuracy and originality.
```

Failure to include this disclosure when applicable is considered a policy violation.

---

## 3. Core Principles

### Originality

Contributors are fully responsible for the **originality, correctness, and integrity** of all submitted work, including code, documentation, and designs.

All dependencies, datasets, and external content must be properly credited and comply with applicable open-source licenses.

---

### Transparency

If an LLM materially influenced a design decision, code snippet, experiment, or analysis, the contributor must:

- Clearly explain the role of the model
- Validate the correctness of the output

> - Mark outputs from closed-source or non-reproducible LLMs as **non-reproducible**

Author: **Clearly explain the role of the model**
Be explicit about what the LLM was used for. Was it generating code, summarising data, drafting text, analysing results? 

**Validate the correctness of the output**
Don't blindly trust LLM outputs. Verify that generated code actually works, that summaries are accurate, that analysis is sound. LLMs can hallucinate, make subtle errors, or produce plausible-sounding nonsense. So please validate the correctness of the output to avoid your proposal being rejected. 

**Mark outputs from closed-source or non-reproducible LLMs as non-reproducible**
If you used a model like ChatGPT or Claude, someone else can't guarantee they'll get the same output, even with the same prompt. Model versions change, weights aren't public, and inference can vary. So if they used a model, they need to explicitly mention it and mark it as non-reproducible.

> - Mark outputs from closed-source or non-reproducible LLMs as **non-reproducible**

Reviewer: @arkid15r Does that cover it or do you have suggested edits?
